### ISSUE TITLE
Ashwalkers no longer spawn with traits nor chosen name if using character slot.

### ISSUE THREAD

**Issue by ARandomAlien** [Skyrat-SS13/Skyrat-TG#11431]
Reporting client version: 514.1578
## Round ID:
4472

## Testmerges:
- [[TM-Only] First attempt at stopping tgchat from eating messages](https://github.com/Skyrat-SS13/Skyrat-tg/pull/8842)
- [[TM?][v0] MekaNIKA cyborg skinline, AKA Tallborgs](https://github.com/Skyrat-SS13/Skyrat-tg/pull/10895)
- [YOU CURRENTLY NEED THIS PR TO RUN THE SERVER WITH A DATABASE CONNECTED ](https://github.com/Skyrat-SS13/Skyrat-tg/pull/10823)
- [[Semi-Modular] Synthetic Rebalance III: A Rope of Sand](https://github.com/Skyrat-SS13/Skyrat-tg/pull/10821)
- [Xenoarch base](https://github.com/Skyrat-SS13/Skyrat-tg/pull/11253)
- [Event voting system, prelude to event planner](https://github.com/Skyrat-SS13/Skyrat-tg/pull/11270)
- [The MODsuit Digital Revolution: Replaces AIs in MODsuits with pAIs](https://github.com/Skyrat-SS13/Skyrat-tg/pull/11135)
- [Black Mesa Source Vox #2](https://github.com/Skyrat-SS13/Skyrat-tg/pull/11393)
- [Revert "[MIRROR] Adds moveloop bucketing, uses queues for the singulo rather then sleeps [MDB IGNORE]"](https://github.com/Skyrat-SS13/Skyrat-tg/pull/11398)

## Reproduction:

Basically, try to spawn as an ashwalker and you'll find that traits and Name's from a character slot ashwalker don't transfer over even though descriptions and the actual looks will

**Comment by malleyeno** [comment]
Can confirm this bug occurs on Skyrat production as specified in the reproduction steps. 

Tried testing this on TG master branch and basil, but those branches don't let users make predefined Ashwalker characters so I think this is a Skyrat specific bug. It might be related to editing mob details, since Skyrat asks users if they want to use a character after the mob is already spawned in (TG doesn't have any pop-up after the mob is spawned).

**Comment by malleyeno** [comment]
Did more testing, and I think I might know what could be causing this (but would like someone who knows what they're doing to look at this first and let me know if I'm off the mark).

Testing on the Skyrat Main, the name change occurs from null to a randomly generated lizard name by the looks of it. Also, this occurs before the spawned character actually gets the Ashwalker antag datum assigned to them. On the logs, this looks like "RealNameHere Name changed from  to Random-Lizard-Name-Hiss" and then "Random-Lizard-Name-Hiss assigned datum /datum/antagonist/ashwalker".

Searching through main, it looks like the only function with similar functionality is in [mining_roles](https://github.com/Skyrat-SS13/Skyrat-tg/blob/3269b61a812a811f6d0c565524780d416bd1e7ac/code/modules/mob_spawn/ghost_roles/mining_roles.dm). Note that there's a random lizard name generator from Null, and that the datum is added after the name change.: 

```
/obj/effect/mob_spawn/ghost_role/human/ash_walker/special(mob/living/carbon/human/spawned_human)
	. = ..()
	spawned_human.fully_replace_character_name(null,random_unique_lizard_name(gender))
	to_chat(spawned_human, "<b>Drag the corpses of men and beasts to your nest. It will absorb them to create more of your kind. Invade the strange structure of the outsiders if you must. Do not cause unnecessary destruction, as littering the wastes with ugly wreckage is certain to not gain you favor. Glory to the Necropolis!</b>")

	spawned_human.mind.add_antag_datum(/datum/antagonist/ashwalker, team)

	ADD_TRAIT(spawned_human, TRAIT_PRIMITIVE, ROUNDSTART_TRAIT)
	spawned_human.remove_language(/datum/language/common)
	team.players_spawned += (spawned_human.key)
	eggshell.egg = null
	QDEL_NULL(eggshell)
```

My guess is that this effect is being triggered despite a character being selected, since it changes datum and Name, but nothing about appearance.

**Comment by malleyeno** [comment]
The naming bug seems to have happened because the name in the character preferences got overwritten by a random lizard name. #11595 attempts to address that.

When I tried to figure out why quirks weren't being put onto the characters, I couldn't find any specific references to them when the mob gets spawned, other than the Skyrat specific changes to the pref transfer proc in mob_spawn.dm (/obj/effect/mob_spawn/ghost_role/special). I'm not sure if the quirks just aren't being applied, or if they get overwritten (just like the name does) down the call stack. It's also possible that Ashwalkers are ineligible for quirks as an intentional design decision.

Update (2022-02-19): My fork of skyrat was off by two weeks, and changes made by Tom have added in quirks for ghost roles. This should address all bugs noticed in this issue.
